### PR TITLE
Closes #2725: Add safe browsing categories to TrackingProtectionPolicy.

### LINKED ISSUE #2725: Unable to trigger Safebrowsing (high risk) errors in Reference Browser / Fenix
When I visit websites in Reference Browser or in Fenix that **should** trigger safebrowsing errors (e.g. Safebrowsing Phishing URI), the error is not intercepted. All medium / low risk errors are caught.

@jonalmeida confirmed this works correctly in sample browser.

### PR DESCRIPTION
**Pull request by Amejia481**
### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### CODE REVIEW

**Comment by Amejia481** [comment]
> I think I prefer the former to add a default method now using CB_DEFAULT and have our all use CB_STRICT. That should cover things for now and we can always expose more methods later. Wdyt?

I like the idea 👍  I will also added some docs to clarify de differences between each other, then we'll have: `DEFAULT = CB_DEFAULT` and `ALL =CB_STRICT`. 

**Side Note**:
We were using [AT_STRICT before](https://github.com/mozilla-mobile/android-components/blob/06bb0891d34679041dc2037e2621ffc20fc7d8bf/components/concept/engine/src/main/java/mozilla/components/concept/engine/EngineSession.kt#L98), because we added `CONTENT + CRYPTOMINING + FINGERPRINTING` to AT_DEFAULT = `AD + ANALYTICS + SOCIAL + CONTENT + TEST`:

[EngineSession.kt#L98](https://github.com/mozilla-mobile/android-components/blob/06bb0891d34679041dc2037e2621ffc20fc7d8bf/components/concept/engine/src/main/java/mozilla/components/concept/engine/EngineSession.kt#L98)
 ```kotlin
 internal const val ALL: Int = AD + ANALYTICS + SOCIAL + CONTENT + TEST + CRYPTOMINING + FINGERPRINTING

 ```

[AT_STRICT](https://searchfox.org/mozilla-central/source/mobile/android/geckoview/src/main/java/org/mozilla/geckoview/ContentBlocking.java#273) and [AT_DEFAULT](https://searchfox.org/mozilla-central/source/mobile/android/geckoview/src/main/java/org/mozilla/geckoview/ContentBlocking.java#266)
 ```java
 /**
     * Block all known trackers.
     * May cause issues with some web sites.
     */
    public static final int AT_STRICT =
        AT_DEFAULT | AT_CONTENT | AT_CRYPTOMINING | AT_FINGERPRINTING;


    /**
     * Block ad, analytic, social and test trackers.
     */
    public static final int AT_DEFAULT =
        AT_AD | AT_ANALYTIC | AT_SOCIAL | AT_TEST;
```

**In this PR we are only adding `SB_ALL`**

[SB_ALL](https://searchfox.org/mozilla-central/source/mobile/android/geckoview/src/main/java/org/mozilla/geckoview/ContentBlocking.java#301)
```java
    /**
     * Block all unsafe sites.
     * Blocks all {@link #SB_MALWARE SB_*} types.
     */
    public static final int SB_ALL =
        SB_MALWARE | SB_UNWANTED | SB_HARMFUL | SB_PHISHING;
```

We're now falling on [CB_STRICT](https://searchfox.org/mozilla-central/source/mobile/android/geckoview/src/main/java/org/mozilla/geckoview/ContentBlocking.java#315) because, we're combining [SB_ALL](https://searchfox.org/mozilla-central/source/mobile/android/geckoview/src/main/java/org/mozilla/geckoview/ContentBlocking.java#301) with [AT_STRICT](https://searchfox.org/mozilla-central/source/mobile/android/geckoview/src/main/java/org/mozilla/geckoview/ContentBlocking.java#273).

```java
    /**
     * Strict content blocking categories.
     * A combination of {@link #AT_STRICT} and {@link #SB_ALL}.
     * May cause issues with some web sites.
     */
    public static final int CB_STRICT = SB_ALL | AT_STRICT;
```

**Comment by Amejia481** [comment]
@csadilek  what do you think about  using `recommended` instead of `default`? II think it conveys better the idea.

**Comment by csadilek** [comment]
@Amejia481 ok `recommended` and `all` sound good too, given what we currently get from GV.

**Comment by Amejia481** [comment]
@csadilek Updated!

File: components/concept/engine/src/main/java/mozilla/components/concept/engine/EngineSession.kt [hunk 1]
```diff
@@ -87,18 +87,72 @@ abstract class EngineSession(
...
+                SAFE_BROWSING_MALWARE + SAFE_BROWSING_UNWANTED + SAFE_BROWSING_HARMFUL + SAFE_BROWSING_PHISHING
+
+            internal const val RECOMMENDED: Int = AD + ANALYTICS + SOCIAL + TEST + SAFE_BROWSING_ALL
+
+            internal const val ALL: Int = RECOMMENDED + CRYPTOMINING + FINGERPRINTING + CONTENT
 
             fun none() = TrackingProtectionPolicy(NONE)
+
+            /**
+             * Strict content blocking categories.
```
Comment (csadilek) [inline components/concept/engine/src/main/java/mozilla/components/concept/engine/EngineSession.kt]: nit: I know it's called content blocking (CB_) in GV, but here we just call things tracking protection policies. I'd change to:

`
Combining the [recommended] categories plus [CRYPTOMINING], [FINGERPRINTING] and [CONTENT]. This is the strictest setting and may cause issues on some web sites.
`

File: components/concept/engine/src/main/java/mozilla/components/concept/engine/EngineSession.kt [hunk 2]
```diff
@@ -87,18 +87,72 @@ abstract class EngineSession(
...
+             * A combination of the [recommended] categories plus [CRYPTOMINING], [FINGERPRINTING] and [CONTENT].
+             * May cause issues with some web sites.
+             */
             fun all() = TrackingProtectionPolicyForSessionTypes(ALL)
+            /**
+             * Recommended content blocking categories.
```
Comment (csadilek) [inline components/concept/engine/src/main/java/mozilla/components/concept/engine/EngineSession.kt]: nit: see above would prefer:

`
Combining the [AD], [ANALYTICS], [SOCIAL], [TEST] categories plus [SAFE_BROWSING_ALL]. This is the recommended setting.
`

**Pull request merged**